connector/failoverconnector/logs.go

```diff
@@ -0,0 +1,48 @@
+package failoverconnector
+
+import (
+	"context"
+	"errors"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logsFailover struct {
+	component.StartFunc
+	component.ShutdownFunc
+
+	config   *Config
+	failover *failoverRouter[consumer.Logs]
+	logger   *zap.Logger
+}
+
+func (f *logsFailover) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeLogs will try to export to the current set priority level and handle failover in the case of an error
+func (f *logsFailover) ConsumeLogs(ctx context.Context, md plog.Logs) error {
+	return nil
+}
+
+func (f *logsFailover) Shutdown(ctx context.Context) error {
+	return nil
+}
+
+func newLogsToLogs(set connector.CreateSettings, cfg component.Config, logs consumer.Logs) (connector.Logs, error) {
+	config := cfg.(*Config)
+	lr, ok := logs.(connector.LogsRouter)
+	if !ok {
+		return nil, errors.New("consumer is not of type LogsRouter")
+	}
+
+	failover := newFailoverRouter(lr.Consumer, config)
```

Comment:
We can save this for a subsequent PR but I think we should validate the contents of all priority levels (i.e. instantiate each consumer) within the constructor, rather than defer until later. This will allow us to fail quickly on invalid input.

Reply:
Yep that's the way it works in the larger PR, I just removed it to avoid adding a mocked method on the failoverrouter

`	if !ok {
		return nil, errors.New("consumer is not of type LogsRouter")
	}

	failover := newFailoverRouter(lr.Consumer, config)
	err := failover.registerConsumers()
	if err != nil {
		return nil, err
	}`

Reply:
Ok, sounds good. This is fine for this PR then.